lib/packages/docutils/rst.nim

```diff
@@ -1937,13 +1948,34 @@ proc parseEnumList(p: var RstParser): PRstNode =
     wildToken: array[0..5, int] = [4, 3, 3, 4, 3, 3]  # number of tokens
     wildIndex: array[0..5, int] = [1, 0, 0, 1, 0, 0]
       # position of enumeration sequence (number/letter) in enumerator
-  result = newRstNodeA(p, rnEnumList)
   let col = currentTok(p).col
   var w = 0
   while w < wildcards.len:
     if match(p, p.idx, wildcards[w]): break
     inc w
   assert w < wildcards.len
+  proc checkAfterNewline(p: RstParser, report: bool): bool =
```

Comment:
Add whitespace around this proc and add a doc comment *why* we need it and why it is so complex.

Reply:
that diff is from https://github.com/nim-lang/Nim/pull/17257 which just got merged; 
regarding why we need it: to give a human understandable warning when https://github.com/nim-lang/Nim/issues/17249 happens
not sure if it can be much simplified though

Reply:
Done.